rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/XtextResourceFormatter.java

```diff
@@ -0,0 +1,80 @@
+package com.regnosys.rosetta.formatting2;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.formatting2.FormatterRequest;
+import org.eclipse.xtext.formatting2.IFormatter2;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionAccess;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionRewriter;
+import org.eclipse.xtext.formatting2.regionaccess.ITextReplacement;
+import org.eclipse.xtext.formatting2.regionaccess.TextRegionAccessBuilder;
+import org.eclipse.xtext.preferences.ITypedPreferenceValues;
+import org.eclipse.xtext.resource.XtextResource;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+public class XtextResourceFormatter implements ResourceFormatterService{
+	@Inject
+	private Provider<FormatterRequest> formatterRequestProvider;
+	
+	@Inject 
+	private Provider<IFormatter2> iFormatter2Provider;
+
+	@Inject
+	private TextRegionAccessBuilder regionBuilder;
+	
+	@Override
+	public void formatCollection(Collection<Resource> resources) {
+	    formatCollection(resources, null);
+	}
```

Comment:
Trivial implementations such as these are usually implemented as a [`default`](https://www.geeksforgeeks.org/default-methods-java/) interface method, so implementers of the interface do not have to write boilerplate code. I would move this one up, together with the two below.